test/libsolidity/syntaxTests/parsing/lexer_numbers_with_underscores_decimal.sol

```diff
@@ -0,0 +1,20 @@
+contract C {
+  function f() public pure {
+    // good
```

Comment:
Maybe better to split the tests into a succeeding and a failing part. Same for the other tests.

Reply:
Sadly not sure what benefits this would give us. But I saw other tests doing that too, yes.

Reply:
It's easier to review and more apparent that everything still works as intended, if something changes. It's difficult to verify the source locations in git diffs, so it may go unnoticed, if a previously succeeding test fails and a previously failing test succeeds (even though that may be unlikely in practice). Originally ``isoltest`` used line numbers, not source location, maybe we should have stayed with that or at least kept them :-D... but it would be a major hassle to change it now :-).

Reply:
Yes please split them. The succeeding ones should never be updated in PRs, unless we make a breaking change, so it will be apparent if anything broke. The failing ones may be updated more frequently, because of better error messages, etc.